pkg/splunk/enterprise/types.go

```diff
@@ -113,6 +113,58 @@ type AppInstallPipeline struct {
 
 	// Refernce to app deploy context
 	appDeployContext *enterpriseApi.AppDeploymentContext
+
+	// reference to PlayBookImpl to run a specific playbook
+	playBookContext PlayBookImpl
```

Comment:
This is something I think I missed earlier, do we need this? I think it brings confusion when we have multiple cluster scoped playbooks. Ex. SHC, ES, ITSI, and additional logic to resolve the same.  On every reconcile entry, the logic has to figure out and prepare the correct playbook context to run the correct playbook, and already all those functions are being methods with those contexts as receivers, so, it is simple just to call them. We can discuss and finalize if we need this.